File: Java/12. JavaOOP_internal_classes/app/src/main/java/com/example/javaoop_internal_classes/MainActivity.java
package com.example.javaoop_internal_classes;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;

// Inner class
// Static inner class
// Local class
// Anonymous class

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Cat cat1 = new Cat(1);
        cat1.talk();

        Cat cat2 = new Cat(3);
        cat2.talk();

        Cat cat3 = new Cat(8);
        cat3.talk();

        Cat.CatMood catMood = new Cat(1).new CatMood();
        Log.i("catMood", String.valueOf(catMood.levelOfMood)); // 100

        Log.i("count", String.valueOf(Cat.countOfInstance)); // 4
        Cat.countOfInstance = 100;
        Log.i("count", String.valueOf(Cat.countOfInstance)); // 100

        Cat.CountResetter countResetter = new Cat.CountResetter();
        countResetter.resetCount();
        Log.i("count", String.valueOf(Cat.countOfInstance)); // 0


        Cat cat5 = new Cat(2);
        cat5.catchMouse(2);
        cat5.catchMouse(10);

        // Anonymous class. Еслм нам надо разово переопределить метод только для этого instance.
        Cat signingCat = new Cat(2) {
            @Override // по сути не обязательно, но лучше писать. Это явно указывает что мы перезаписываем уже существующий метод
            public void talk() {
                Log.i("talk()", "I'm signing instance of cat");
            }
        };
        signingCat.talk();

    }
}

File: Kotlin/2. Magic8Ball/app/src/main/java/com/android/magic8ball/MainActivity.kt
package com.android.magic8ball

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import java.util.*

class MainActivity : AppCompatActivity() {
    var askList = listOf<String>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val askMeButton: Button = findViewById(R.id.askButton);
        askList = listOf(
            "It is certain", "It is decidedly so", "Without a doubt",
            "Yes - definitely", "You may rely on it", "As I see it, yes",
            "Most likely", "Outlook good", "Yes", "Signs point to yes",
            "Reply hazy, try again", "Ask again later", "Better not tell you now",
            "Cannot predict now", "Concentrate and ask again", "Don't count on it",
            "My reply is no", "My source say no", "Outlook noe so good",
            "Very doubtful"
        )

        askMeButton.setOnClickListener { shakeBall() }
    }

    private fun shakeBall() {
        val randomChoice = Random().nextInt(askList.size)
        val textAnswer:TextView = findViewById(R.id.answerText)
        textAnswer.text = askList[randomChoice]
    }
}